src/components/basic/DlWidget/utils.ts

```diff
@@ -78,7 +77,23 @@ export function swapElemensInMatrix(
         removedElement[0]
     )
 
-    return isTooLarge(newLayout, maxElements) ? layout : newLayout
+    return isTooLarge(newLayout, maxElements)
+        ? moveElementsToNextIndex(newLayout, maxElements)
+        : newLayout
+}
+
+function moveElementsToNextIndex(array: number[][], maxElements: number) {
```

Comment:
@vadim-popa-opentech i dont like this approach where we modify the array we receive. i would much rather have that you go over the array in a normal for loop and add the relevant data to the new generated array. 

Refrain from modifying parameters as it can cause confusion and unexpected behavior.